v2/api/network/v1api20220701/structure.txt

```diff
@@ -2,6 +2,1207 @@
 github.com/Azure/azure-service-operator/v2/api/network/v1api20220701
 ├── APIVersion: Enum (1 value)
 │   └── "2022-07-01"
+├── ApplicationGateway: Resource
+│   ├── Owner: github.com/Azure/azure-service-operator/v2/api/resources/v1apiv20191001.ResourceGroup
+│   ├── Spec: Object (37 properties)
+│   │   ├── AuthenticationCertificates: Object (2 properties)[]
+│   │   │   ├── Data: *string
+│   │   │   └── Name: *string
+│   │   ├── AutoscaleConfiguration: *Object (2 properties)
+│   │   │   ├── MaxCapacity: Validated<*int> (1 rule)
+│   │   │   │   └── Rule 0: Minimum: 2
+│   │   │   └── MinCapacity: Validated<*int> (1 rule)
+│   │   │       └── Rule 0: Minimum: 0
+│   │   ├── AzureName: string
+│   │   ├── BackendAddressPools: Object (2 properties)[]
+│   │   │   ├── BackendAddresses: Object (2 properties)[]
+│   │   │   │   ├── Fqdn: *string
+│   │   │   │   └── IpAddress: *string
+│   │   │   └── Name: *string
+│   │   ├── BackendHttpSettingsCollection: Object (14 properties)[]
+│   │   │   ├── AffinityCookieName: *string
+│   │   │   ├── AuthenticationCertificates: Object (1 property)[]
+│   │   │   │   └── Reference: *genruntime.ResourceReference
```

Comment:
It would be nice to determine what sort of resource this is, and if we should also add support for it to our list.

Reply:
Ah, it might actually be created by the AuthenticationCertificates section of this resource itself. That will make referring to it a bit annoying as it won't have its own Kubernetes ID. Not that there's much we can do about that.

Reply:
it is quite common in app gateway that internal app gateway resource are linked IP, in this case the BackendHttpSettingsCollection can have a reference to an AuthenticationCertificates, thta is itself a sub resource in application gateway